### PR TITLE
Pytestify tztrout

### PR DESCRIPTION
**Pull request by eengoron**
This PR pytest-ifys `tztrout` and gets rid of `unittest.` I did this as a prerequisite to #12, so we won't have to write any more unittests when testing our new method for handling zipcodes.

I also realized that the readme incorrectly uses `tz_trout` instead of `tztrout` so I switched it to be correct in the readme. 

In trying to get `pytest` to pass on CircleCI I had some issues running the pytest command. So, I reconfigured the `config.yaml` file to build this package in a way similar to our other repos like [limitlion](https://github.com/closeio/limitlion/blob/master/.circleci/config.yml) and [tasktiger](https://github.com/closeio/tasktiger/blob/master/.circleci/config.yml)

As a side note, `black`, `flake8`, and `isort` should be added to CircleCI as well, but that can be handled in a follow-up PR. I blackened the `test_tztrout.py` here simply for readability.

### CODE REVIEW

File: tests/test_tztrout.py [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+            ('+1 (650) 333 4444', ['America/Los_Angeles']),
+            ('+48 601 941 311', ['Europe/Warsaw']),
+        ],
+    )
+    def test_ids_for_phone(self, phone, tz_ids):
+        ids = tztrout.tz_ids_for_phone(phone)
+        assert ids == tz_ids
+
+    @pytest.mark.parametrize(
+        'country, state, city, zipcode, tz_ids, assume_any_tz',
```
Comment (wojcikstefan) [inline tests/test_tztrout.py]: 1. `tz_ids` -> `expected_tz_ids`.
1. I think `is_exact_match` would be a more easily understandable name vs `assume_any_tz`.

File: tests/test_tztrout.py [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+        ],
+    )
+    def test_major_cities_us_ca(
+        self,
+        country,
+        state,
+        city,
+        phones,
+        tz_name,
+        assert_ids_equal,
```
Comment (wojcikstefan) [inline tests/test_tztrout.py]: Is `assert_ids_equal` and `is_exact_match` again?

File: tests/test_tztrout.py [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+        """ Make sure all the major cities in the United States and Canada match the right time zone
+        (and the right time zone only). """
+        ids = tztrout.tz_ids_for_address(country, state=state, city=city)
+        for phone in phones:
+            ids2 = tztrout.tz_ids_for_phone(phone)
+            if assert_ids_equal:
+                assert ids == ids2
+            else:
+                self.assert_only_one_tz(ids2, tz_name, us_ca_tz_names)
+        self.assert_only_one_tz(ids, tz_name, us_ca_tz_names)
```
Comment (wojcikstefan) [inline tests/test_tztrout.py]: This test seems to be testing two things (`tz_ids_for_address` and `tz_ids_for_phone`). They should both be tested separately.

File: tests/test_tztrout.py [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+
+    def test_canada_without_state_info(self):
+        ids = tztrout.tz_ids_for_address('CA')
```
Comment (wojcikstefan) [inline tests/test_tztrout.py]: I think we should break these tests into classes per method, so you'd have e.g. `TestTzIdsForAddress` that groups together all the tests verifying the behavior of `tz_ids_for_address`, another test class for `tz_ids_for_phone`, another one for `local_time_for_address`, and so on...

File: tests/test_tztrout.py [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+    def test_major_cities_australia(
+        self, state, city, phones, tz_name, au_tz_names
+    ):
+        """ Make sure all the major cities in Australia match the right time zone
+        (and the right time zone only). """
+        ids = tztrout.tz_ids_for_address('AU', state=state, city=city)
+        for phone in phones:
+            ids2 = tztrout.tz_ids_for_phone(phone)
+            assert ids == ids2
+        self.assert_only_one_tz(ids, tz_name, au_tz_names)
```
Comment (wojcikstefan) [inline tests/test_tztrout.py]: Again, we're testing two different things here. Doesn't make sense to group them together IMO.

File: tests/test_tztrout.py [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+            (20, '9am', '5pm', [[-11 * 60, -3 * 60], [13 * 60, 14 * 60]]),
+            (20, '9:00', '17:00', [[-11 * 60, -3 * 60], [13 * 60, 14 * 60]]),
+        ],
+    )
+    def test_offset_ranges(self, hour_now, range_start, range_end, result):
+        FakeDateTime.set_utcnow(datetime.datetime(2013, 1, 1, hour_now))
+        offset_ranges = tztrout.offset_ranges_for_local_time(
+            range_start, range_end
+        )
+        assert offset_ranges == result
```
Comment (wojcikstefan) [inline tests/test_tztrout.py]: It's hard to gauge just by looking at the params and the test code – how does this test work? And what is it asserting?

File: tests/test_tztrout.py [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+        for other_name in tz_names_copy:
+            assert not (set(tztrout.tz_ids_for_tz_name(other_name)) & set(ids))
+
+    @pytest.fixture
+    def us_ca_tz_names(self):
+        return ['PT', 'MT', 'CT', 'ET', 'AT']
+
+    @pytest.fixture
+    def au_tz_names(self):
+        return ['AWT', 'ACT', 'AET']
```
Comment (wojcikstefan) [inline tests/test_tztrout.py]: These might as well be constants, they don't need to be fixtures.

File: tests/test_tztrout.py [hunk 8]
```diff
@@ -0,0 +1,433 @@
...
+    def assert_only_one_tz(self, ids, tz_name, tz_names):
+        """ Assert that a given set of timezone ids only matches one tz name
+         in a given set of tz names
+         """
+        tz_names_copy = tz_names[:]
+        assert tz_name in tz_names_copy
+        tz_names_copy.remove(tz_name)
+        assert set(tztrout.tz_ids_for_tz_name(tz_name)) & set(ids)
+        for other_name in tz_names_copy:
+            assert not (set(tztrout.tz_ids_for_tz_name(other_name)) & set(ids))
```
Comment (wojcikstefan) [inline tests/test_tztrout.py]: This is quite a convoluted assertion (and I know it's been here before, but now seems to be the time to untangle this web). What are we actually using it for? Is there a less complex way to assert the correctness of whatever we're testing?

File: requirements_test.txt [hunk 9]
```diff
@@ -0,0 +1,2 @@
+pytest
+mock
```
[the remark this answers is not in the document]
Comment (eengoron) [reply requirements_test.txt]: done in 2e562eb

File: tests/test_tztrout.py (hunk 1 above)
Comment (eengoron) [reply tests/test_tztrout.py]: Done in e2b8675

File: tests/test_tztrout.py (hunk 2 above)
Comment (eengoron) [reply tests/test_tztrout.py]: I got rid of the need for this variable when splitting out phone and address tests for major cities: 20a80d7

File: tests/test_tztrout.py (hunk 3 above)
Comment (eengoron) [reply tests/test_tztrout.py]: Done in 20a80d7

File: tests/test_tztrout.py (hunk 4 above)
Comment (eengoron) [reply tests/test_tztrout.py]: All of my latest commits made this happen :)

File: tests/test_tztrout.py (hunk 6 above)
Comment (eengoron) [reply tests/test_tztrout.py]: I switched around the variable names to make them clearer, and added a comment for what the test is doing. Let me know if that makes things clearer: 6b3c8ce

File: tests/test_tztrout.py (hunk 8 above)
Comment (eengoron) [reply tests/test_tztrout.py]: I rewrote this method in 22a8186 to use less complexity and more straightforward variable names and comments.

File: tests/test_tztrout.py (hunk 7 above)
Comment (eengoron) [reply tests/test_tztrout.py]: done in 30e01b7

**Comment by eengoron** [comment]
@wojcikstefan -- Feel free to look at this one before #25. I'll rebase and fix #25 after this one is merged :)

File: tests/test_tztrout.py (hunk 8 above)
Comment (wojcikstefan) [reply tests/test_tztrout.py]: I still dislike that we test things in this way (we should separately test the return values of `tz_ids_for_tz_name`, `tz_ids_for_phone`, and `tz_ids_for_address`), but that can be a future improvement and at least what we have is decently readable now. Well done!

**Pull request merged**